tests/core/test_datamodules.py

```diff
@@ -40,62 +40,62 @@
 @mock.patch("pytorch_lightning.trainer.trainer.Trainer.node_rank", new_callable=PropertyMock)
 @mock.patch("pytorch_lightning.trainer.trainer.Trainer.local_rank", new_callable=PropertyMock)
 def test_can_prepare_data(local_rank, node_rank):
```

Comment:
improved test: it is better to check if the prepare_data hook is called directly instead of relying on a "correct" DataModule implementation for checking `dm.random_full` attribute

Reply:
furthermore, in a follow up this test should be moved to test_dataconnector.py because that's what it really is testing.

Reply:
you can move it in this one?

Reply:
No! This is against the rules. It would not be possible (or very hard) for reviewers to review the changes